sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -522,36 +522,58 @@ class Dataset[T] private[sql](
   // scalastyle:on println
 
   /**
-   * Prints the plans (logical and physical) to the console for debugging purposes.
+   * Prints the plans (logical and physical) with a format specified by a given explain mode.
    *
    * @group basic
-   * @since 1.6.0
+   * @since 3.0.0
    */
-  def explain(extended: Boolean): Unit = {
+  def explain(mode: ExplainMode): Unit = {
```

Comment:
How about retain the old api and add a new api ?

Reply:
@ulysses-you . @maropu already did. Please see [line 564](https://github.com/apache/spark/pull/26829/files#diff-7a46f10c3cedbf013cf255564d9483cdR564).

Reply:
Yea, we should keep this. Thanks for the comment, @dongjoon-hyun

Reply:
Oh, I see it.